server/src/store.ts

```diff
@@ -1,16 +1,57 @@
 import Vue from 'vue';
 import Vuex from 'vuex';
+import axios from 'axios';
+// import VueResource from 'vue-resource';
 
-Vue.use(Vuex);
+Vue.use( Vuex );
 
 export default new Vuex.Store({
-  state: {
+  // Same as 'data' for a Vue component, except accessable by all components.
+  state: { // {{{
+    login: {
+      username: '',
+      password: '',
+      response: {},
+    },
+  }, // }}}
 
-  },
-  mutations: {
+  // Mutations commit and track state changes.
+  mutations: { // {{{
+    setLoginPassword( state, password ) {
+      state.login.password = password;
```

Comment:
Although this pattern is pretty verbose, lets keep going with it.

Reply:
This is the pattern to mutate data in the store. Not much we can do about it. We can opt not to put things in the store if you want.